File: app/src/main/java/com/fstyle/structure_android/screen/main/MainContract.java
package com.fstyle.structure_android.screen.main;

import com.fstyle.structure_android.data.model.User;
import com.fstyle.structure_android.data.source.remote.config.error.BaseException;
import com.fstyle.structure_android.screen.BasePresenter;
import com.fstyle.structure_android.utils.rx.BaseSchedulerProvider;

import java.util.List;

/**
 * Created by le.quang.dao on 10/03/2017.
 */

public interface MainContract {

    /**
     * View
     */
    interface View {

        String getKeyword();

        String getLimitNumber();

        // Response from Presenter
        void onSearchError(BaseException throwable);

        void onSearchUsersSuccess(List<User> users);

        void onInvalidKeyWord(String errorMsg);

        void onInvalidLimitNumber(String errorMsg);

        void onDataValid();
    }

    /**
     * Presenter
     */
    interface Presenter extends BasePresenter<View> {

        void setSchedulerProvider(BaseSchedulerProvider schedulerProvider);

        void validateDataInput();

        void searchUsers();
    }
}

File: app/src/main/java/com/fstyle/structure_android/screen/main/MainContract.kt
package com.fstyle.structure_android.screen.main

import com.fstyle.structure_android.data.model.User
import com.fstyle.structure_android.data.source.remote.api.error.BaseException
import com.fstyle.structure_android.screen.BasePresenter
import com.fstyle.structure_android.screen.BaseViewModel
import com.fstyle.structure_android.utils.rx.BaseSchedulerProvider

/**
 * Created by le.quang.dao on 10/03/2017.
 */

interface MainContract {

  /**
   * ViewModel
   */
  interface ViewModel : BaseViewModel {
    fun onRequestServerError(e: BaseException)

    fun onSearchUsersSuccess(users: List<User>)

    fun onInvalidKeyWord(errorMsg: String?)

    fun onInvalidLimitNumber(errorMsg: String?)

    fun onShowProgressBar()

    fun onHideProgressBar()
  }

  /**
   * Presenter
   */
  interface Presenter : BasePresenter<ViewModel> {

    fun setSchedulerProvider(schedulerProvider: BaseSchedulerProvider)

    fun validateKeywordInput(keyword: String?)

    fun validateLimitNumberInput(limit: String?)

    fun validateDataInput(keyword: String?, limit: String?): Boolean

    fun searchUsers(keyWord: String?, limit: Int)
  }
}
